<launch>  
    <arg name="world_name" default="$(find arm_robots)/world/val_plant.world"/>
    <arg name="gazebo_home_directory" default="/home/miguel"/>

    <!-- Setting return to default_automatically will make the robot plan to pose to a default position. When launching, the robot will move (without human verification).
    Also after the robot grabs/selects it will return to the default pose. Only set to true if you are confident about your world environment and it won't collide. 
    Since this is for a simulation, we will return to default automatically.-->
    <param name="return_to_default_automatically" value="True"/>
    <param name="default_pose" value="bent"/> <!-- Default pose is the name specified in the hdt_michigan_moveit/config/hdt-michigan.srdf file-->

    <!-- Launch Val -->
    <include file="$(find plant_selector)/launch/include/val_off_husky.launch">
        <arg name="gazebo_home" value="$(arg gazebo_home_directory)"/>
        <arg name="world_name" value="$(find plant_selector)/worlds/val_plant.world"/>
    </include>

    <!-- Launch Basic Plant Selector Stuff -->
    <include file="$(find plant_selector)/launch/include/plant_selector.launch">
        <arg name="rviz_file" value="$(find plant_selector)/rviz/val.rviz"/>
        <arg name="publish_zed_urdf" value="false"/>
    </include>

    <node name="repetitive_plant_extraction" pkg="plant_selector" type="repetitive_plant_extraction.py"
          output="screen"/>
</launch>

